brainvisa/toolboxes/spm/processes/spm12/util/SPM12composition_generic.py

```diff
@@ -192,12 +196,22 @@ def execution(self, context):
             deformation_element.flow_direction = getattr(self, f'direction_{i}')
             deformation_element.time_step = getattr(self, f'time_step_{i}')
             deformation_element.setDartelTemplatePath(getattr(self, f'dartel_template_{i}').fullPath())
+
         elif deformation_type == 'deformation':
             deformation_element = composition.DeformationField()
-            deformation_element.deformation_field_path = getattr(self, f'deformation_field_{i}').fullPath()
+            deformation_file = getattr(self, f'deformation_field_{i}').fullPath()
+            if deformation_file.endswith('.gz'):
+                src_gz = shutil.copy(deformation_file, temp_directory.fullPath())
+                context.system('gunzip', src_gz)
+                source = '.'.join(src_gz.split('.')[:-1])
```

Comment:
since you are doing something a bit by hand, maybe simply `src_gz[:-3]` would be simpler?

Reply:
FYI: if you do not know in advance the exact extension, it is possible to remove what is after the last point with `src_gz.rsplit('.', 1)[0]`. But here, @clarafischer suggestion is good.

Reply:
Thanks for mention it @sapetnioc , I am not used to use this method :+1:

Reply:
Thanks for you mention @sapetnioc, I am not used to call this method :+1: